tools/tools.go

```diff
@@ -211,9 +212,14 @@ func findArguments(tool api.Tool, buffer []byte) (map[string]any, int) {
 		switch obj := obj.(type) {
 		case map[string]any:
 			valid := true
+			properties := getProperties(tool.Function.Parameters)
+			if properties == nil {
+				return nil
+			}
+
 			// check if all keys in the object exist in the tool's parameters
 			for key := range obj {
-				if _, exists := tool.Function.Parameters.Properties[key]; !exists {
+				if _, exists := properties[key]; !exists {
```

Comment:
Yeah I think we should have a type for it instead to not have to subscript like so